### PR TITLE
Added EG tenet to check for underscores in names

### PR DESCRIPTION
**Pull request by Daanikus**
Tenet for https://golang.org/doc/effective_go.html#mixed-caps

### CODE REVIEW

**Comment by mullikine** [comment]
LGTM but need update expected.json here is mine https://ptpb.pw/nbpb.json

File: tenets/codelingo/effective-go/underscores-in-name/example.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Package main used for testing of tenet
+package main
+
+func main() {
+	var a = 15
+	a_bad_name := 20
+	var another_bad_name = "hello"
```
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /ignore

File: tenets/codelingo/effective-go/underscores-in-name/example.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+func aFunction() {
+	fmt.Println("This name is fine")
+}
+
+func a_bad_function() {
+	fmt.Println("Needs a name change")
+}
+
+// Bad_Interface used as example
+type Bad_Interface interface {
```
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /ignore

File: tenets/codelingo/effective-go/underscores-in-name/example.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	a_bad_name := 20
+	var another_bad_name = "hello"
+	aGoodVar := 2
+}
+
+func aFunction() {
+	fmt.Println("This name is fine")
+}
+
+func a_bad_function() {
```
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /ignore

File: tenets/codelingo/effective-go/underscores-in-name/example.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /review

File: tenets/codelingo/effective-go/underscores-in-name/example.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /ignore

File: tenets/codelingo/effective-go/underscores-in-name/example.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /ignore

File: tenets/codelingo/effective-go/underscores-in-name/example.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Daanikus) [reply tenets/codelingo/effective-go/underscores-in-name/example.go]: /ignore

**Pull request merged**